os/arch/arm/src/armv7-r/arm_schedyield.c

```diff
@@ -0,0 +1,174 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+
+#include <tinyara/sched.h>
+#include <tinyara/ttrace.h>
+#ifdef CONFIG_TASK_SCHED_HISTORY
+#include <tinyara/debug/sysdbg.h>
+#endif
+
+#include "sched/sched.h"
+#include "up_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: arm_schedyield
+ *
+ * Description:
+ *   Called when current task is willing to
+ *   yield cpu resource to other ready-to-run task.
+ *   context switch would occur in below case:
+ *
+ *   1) The priority of the currently running task is either equal or more than other ready to run task
+ *
+ * Inputs:
+ *   tcb: The TCB of the task that has been reprioritized
+ *
+ ****************************************************************************/
+
+void up_schedyield(struct tcb_s *rtcb)
```

Comment:
I'm not sure the param `rtcb` is needed.

Reply:
Yeah you're right. That is unnecessary so that I removed somewhere but I missed here and header.